src/transformers/modeling_tf_transfo_xl.py

```diff
@@ -745,13 +764,24 @@ def __init__(self, config):
         self.sample_softmax = config.sample_softmax
         # use sampled softmax
         if config.sample_softmax > 0:
-            raise NotImplementedError
+            self.out_layer = TFTransfoXLLMHead(config, self.transformer.word_emb.weight, name="out_layer")
+            self.sampler = TFLogUniformSampler(config.vocab_size, config.sample_softmax)
         # use adaptive softmax (including standard softmax)
         else:
             self.crit = TFAdaptiveSoftmaxMask(
                 config.vocab_size, config.d_embed, config.d_model, config.cutoffs, div_val=config.div_val, name="crit"
             )
 
+    def get_output_embeddings(self):
+        """ Double-check if you are using adaptive softmax.
+        """
+        if self.sample_softmax > 0:
+            return self.out_layer
+        else:
+            if len(self.crit.out_layers) > 0:
+                return self.crit.out_layers[-1]
+            return None
```

Comment:
At initialization of `TFTransfoXLLMHead`, `self.crit.out_layers` is an empty list -> therefore only return last layer if it has one.